spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPersistentEntityImpl.java

```diff
@@ -182,9 +193,39 @@ public boolean hasIdProperty() {
 	public void verify() {
 		super.verify();
 		verifyPrimaryKeysConsecutive();
+		verifyOneToManyProperties();
 		verifyEmbeddedColumnNameOverlap(new HashSet<>(), this);
 	}
 
+	private void verifyOneToManyProperties() {
+		doWithChildCollectionProperties(spannerPersistentProperty -> {
+			// getting the inner type will throw an exception if the property isn't a
+			// collection.
+			Class childType = spannerPersistentProperty.getColumnInnerType();
+			SpannerPersistentEntityImpl childEntity = (SpannerPersistentEntityImpl)
+					this.spannerMappingContext.getPersistentEntity(childType);
+			SpannerPersistentProperty[] primaryKeyProperties = getPrimaryKeyProperties();
+			SpannerPersistentProperty[] childKeyProperties = childEntity
+					.getPrimaryKeyProperties();
+			if (primaryKeyProperties.length >= childKeyProperties.length) {
+				throw new SpannerDataException(
+						"A child table must contain the primary key columns of its "
+								+ "parent in the same order starting the first "
+								+ "column with additional" + " key columns after.");
+			}
+			for (int i = 0; i < primaryKeyProperties.length; i++) {
+				SpannerPersistentProperty parentKey = primaryKeyProperties[i];
+				SpannerPersistentProperty childKey = childKeyProperties[i];
+				if (!parentKey.getColumnName().equals(childKey.getColumnName())
+						|| !parentKey.getType().equals(childKey.getType())) {
+					throw new SpannerDataException(
+							"The child primary key column at position " + (i + 1)
```

Comment:
Why not report zero-based index? You can also name the property and its type for both parent and child to aid debugging.

Reply:
The annotations where the user specify the key order starts with 1 and increases from there, so I wanted to show this number that the user input in the exception. I also wanted to say the number instead of the names of the 2 properties because I felt it was more concise and exact to say that what we detected was wrong was based on the position, and not that the names mismatched. This sounds like a semantic difference, but I think it is less confusing and doesn't imply to the user that there could be a typo or two real columns but just ordered wrong.

Reply:
Ok on the 1-based index, but what's the wrong with including both, the position and the name/type that was found to be mismatched in the exception? As a developer, I would want as much information in the exception as possible to aid debugging.

Reply:
gotcha, will add.